scripts/aptly/aptly_importer.py

```diff
@@ -0,0 +1,338 @@
+            self.filter_formula = self.reprepro2aptly.convert(
+                self.config['filter_formula'])
+            self.method = self.config['method']
+            self.name = self.config['name']
+            self.suites = self.config['suites']
+        except KeyError as e:
+            self.__error(f"{e} key was not found in file {input_file}")
+
+    def __error(self, msg):
+        print(f"Configuration file error: {msg} \n", file=stderr)
+        exit(-1)
+
+    def __load_config_file(self, config_file_path):
+        fn = Path(__file__).parent / config_file_path
+        try:
+            with open(str(fn), 'r') as stream:
+                config = yaml.safe_load(stream)
+                return config
+        except yaml.YAMLError as exc:
+            self.__error(f"yaml parsing error {exc}")
+        except FileNotFoundError as e:
+            self.__error(f"not found {config_file_path}")
+
+
+class UpdaterManager():
+    def __init__(self, input_file, debug=False, aptly_config_file=None):
+        self.aptly = Aptly(debug,
+                           config_file=aptly_config_file)
+        self.config = UpdaterConfiguration(input_file)
+        self.debug = debug
+        self.snapshot_timestamp = None
+
+    def __assure_aptly_mirrors_do_not_exist(self):
+        self.__log('Checking that aptly mirrors from config filename do not exist')
+        for dist in self.config.suites:
+            mirror_name = self.__get_mirror_name(dist)
+            if self.aptly.exists(Aptly.ArtifactType.MIRROR, mirror_name):
+                self.__error(f"mirror {mirror_name} exists. Refuse to create mirrors")
+        self.__log_ok('no conflict in mirrors name')
+
+    def __create_aptly_mirror(self, distribution):
+        assert(self.config)
+        self.__log(f"Creating aptly mirror for {distribution}")
+        mirror_name = self.__get_mirror_name(distribution)
+        self.aptly.run(['mirror', 'create', '-with-sources',
+                        f"-architectures={','.join(self.config.architectures)}",
+                        f"-filter={self.config.filter_formula}",
+                        mirror_name,
+                        self.config.method,
+                        distribution,
+                        self.config.component])
+        self.aptly.run(['mirror', 'update', mirror_name])
+        self.__log_ok(f"mirror {mirror_name} created")
+
+    def __create_aptly_snapshot(self, distribution):
+        self.__log('Creating an aptly snapshot from local aptly repository')
+        self.aptly.run(['snapshot', 'create', self.__get_snapshot_name(distribution),
+                        'from', 'mirror', self.__get_mirror_name(distribution)])
+        self.__log_ok(f"snapshot {self.__get_snapshot_name(distribution)} created from mirror {self.__get_mirror_name(distribution)}")
+
+    def __error(self, msg):
+        print(f"Update Manager error: {msg} \n", file=stderr)
+        exit(-1)
```

Comment:
Does it make sense to use a low valued exit code rather than -1/255?

Reply:
Used 1 for generic errors, 2 for config file errors. 7e74b46